src/luv.c

```diff
@@ -508,20 +508,54 @@ static void luv_handle_init(lua_State* L) {
   lua_setfield(L, LUA_REGISTRYINDEX, "uv_stream");
 }
 
-LUALIB_API lua_State* luv_state(uv_loop_t* loop) {
-  return (lua_State*)loop->data;
+static const char* luv_loop_key = "uv_loop";
+static const char* luv_state_key = "main_thread";
```

Comment:
This should be changed to `"luv_main_thread"` to make sure it never collides with anything else.